packets/game_state_packet.cpp

```diff
@@ -19,46 +19,52 @@ size_t GameStatePacket::calculateSize() const {
 // Ensure outData is large enough to store all the data that is intended to be serialized.
 void GameStatePacket::serialize(const GameStatePacket& packet, char*& outData) {
     char* temp = outData;
+    int size = 0;
     
     // Serialize number of players
     size_t numPlayers = packet.state.players.size();
     memcpy(temp, &numPlayers, sizeof(numPlayers));
     temp += sizeof(numPlayers);
+    size += sizeof(numPlayers);
 
-    // Serialize each player's state including x, y, z, orientation, and score
+    // Serialize each player's state including 16 floats of world matrix, and score
     for (const PlayerState& player : packet.state.players) {
-        memcpy(temp, &player.x, sizeof(player.x));
-        temp += sizeof(player.x);
-        memcpy(temp, &player.y, sizeof(player.y));
-        temp += sizeof(player.y);
-        memcpy(temp, &player.z, sizeof(player.z));
-        temp += sizeof(player.z);
-        memcpy(temp, &player.orientation, sizeof(player.orientation));
-        temp += sizeof(player.orientation);
+        for (int i = 0; i < 4; i++) {
+            for (int j = 0; j < 4; j++) {
+                memcpy(temp, &player.world[i][j], sizeof(player.world[i][j]));
```

Comment:
I think the nested for loop (line 32-34) can be simplifed to memcpy(temp, &player.world, sizeof(player.world)) ?

Reply:
I don't think glm::mat4s are guaranteed to be contiguous in memory, I need to do a bit more research for that